packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -982,6 +985,33 @@ abstract class ResidentRunner {
     }
   }
 
+  io.HttpServer _devtoolsServer;
+
+  Future<void> launchDevTools() async {
```

Comment:
For testing: what I would do is give the ResidentRunner an optional constructor argument:

```
Future<io.HttpServer> Function({bool enableStdinCommands}) serveDevTools
```

Then in the launchDevTools method do something like:

```
await (serveDevTools ?? devtools_server.serveDevTools)(enabledStinCommands: false)
```

That way we can test it without adding more to the context in the short term

Reply:
After further thought, not much of the behavior of the devtools server and its client in the browser can be tested. Instead, I added a test to `terminal_handler_test.dart` that ensures that `ResidentRunner.launchDevTools` is called when we get 'v' on the terminal.

Reply:
Yeah seems reasonable to me